### PR TITLE
[GNA] Fix sporadic failures in IE tests

### PR DESCRIPTION
**Pull request by kmagiers**

### CODE REVIEW

File: inference-engine/src/gna_plugin/gna_device.cpp [hunk 1]
```diff
@@ -376,6 +378,11 @@ void GNADeviceHelper::close() {
     GNADeviceClose(nGNAHandle);
     nGNAHandle = 0;
 #else
+    if (wasAnyRequestScheduled) {
+        try {
+            wait(lastSheduledRequestId);
```
Comment (kbruniec) [inline inference-engine/src/gna_plugin/gna_device.cpp:383]: this should not be called if Gna2RequestWait was already called and got status of completition.

File: inference-engine/src/gna_plugin/gna_device.cpp [hunk 2]
```diff
@@ -80,6 +80,8 @@ uint32_t GNADeviceHelper::propagate(const uint32_t requestConfigId, Gna2Accelera
     checkGna2Status(status1);
     const auto status2 = Gna2RequestEnqueue(requestConfigId, &reqId);
     checkGna2Status(status2);
+    wasAnyRequestScheduled = true;
```
Comment (kbruniec) [inline inference-engine/src/gna_plugin/gna_device.cpp:83]: consider to falsify this flag if gna2requestwait was called and got 'full completition' status.

Comment (kmagiers) [reply inference-engine/src/gna_plugin/gna_device.cpp:83]: it is indeed a mistake, it was supposed to be false

File: inference-engine/src/gna_plugin/gna_device.cpp (hunk 1 above)
Comment (kmagiers) [reply inference-engine/src/gna_plugin/gna_device.cpp:383]: are there any issues with it (except exception?) keeping track of all calls may waste memory

File: inference-engine/src/gna_plugin/gna_device.cpp [hunk 3]
```diff
@@ -376,6 +378,11 @@ void GNADeviceHelper::close() {
     GNADeviceClose(nGNAHandle);
     nGNAHandle = 0;
 #else
+    if (wasAnyRequestScheduled) {
+        try {
+            wait(lastSheduledRequestId);
+        } catch (...) {}
```
Comment (kbruniec) [inline inference-engine/src/gna_plugin/gna_device.cpp:384]: at least I would log something, and not hide every exception completelly

File: inference-engine/src/gna_plugin/gna_device.cpp (hunk 1 above)
Comment (kmagiers) [reply inference-engine/src/gna_plugin/gna_device.cpp:383]: Or actually i can track only single wait

File: inference-engine/src/gna_plugin/gna_device.hpp [hunk 4]
```diff
@@ -49,7 +49,8 @@ class GNADeviceHelper {
     uint64_t instrumentationResults[TotalGna2InstrumentationPoints] = {};
     uint64_t instrumentationTotal[TotalGna2InstrumentationPoints] = {};
     uint32_t instrumentationConfigId = 0;
-
+    uint32_t lastSheduledRequestId;
```
[the remark this answers is not in the document]
Comment (kmagiers) [reply inference-engine/src/gna_plugin/gna_device.hpp]: done

File: inference-engine/src/gna_plugin/gna_device.cpp (hunk 3 above)
Comment (kmagiers) [reply inference-engine/src/gna_plugin/gna_device.cpp]: done

File: inference-engine/src/gna_plugin/gna_device.cpp (hunk 1 above)
Comment (kmagiers) [reply inference-engine/src/gna_plugin/gna_device.cpp]: New implementation takes different approach, which solves this issue

**Review by kbruniec — APPROVED** [review]
IMO Can be merged

File: inference-engine/src/gna_plugin/gna_device.cpp [hunk 5]
```diff
@@ -372,6 +376,14 @@ void GNADeviceHelper::open(uint8_t n_threads) {
 }
 
 void GNADeviceHelper::close() {
+    auto requestsToClose = unwaitedRequestIds;
```
Comment (dorloff) [inline inference-engine/src/gna_plugin/gna_device.cpp:379]: Why do we need this additional variable?

Comment (kmagiers) [reply inference-engine/src/gna_plugin/gna_device.cpp:379]: unwaitedRequestsIds is modified during for loop restroying the validity of iterator

**Pull request merged**